src/NLog/Targets/WebServiceTarget.cs

```diff
@@ -180,6 +180,7 @@ public WebServiceTarget(string name) : this()
         /// <docgen category='Web Service Options' order='10' />
         public string XmlRootNamespace { get; set; }
 
+        private readonly Dictionary<HttpWebRequest, KeyValuePair<DateTime,AsyncContinuation>> _pendingRequests = new Dictionary<HttpWebRequest, KeyValuePair<DateTime,AsyncContinuation>>();
```

Comment:
Another way would be using a linkedlist and send the info with the `AsyncContinuation` (if possible)?